src/Nethermind/Nethermind.Runner/Nethermind.Runner.csproj

```diff
@@ -84,11 +84,36 @@
       <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
     </None>
     <None Remove="keystore\**" />
+    <None Remove="scripts\ridgeLine.ts" />
+    <None Remove="scripts\treeMap.ts" />
+    <None Remove="scripts\types.ts" />
+    <None Remove="scripts\utilities.ts" />
+    <None Include="package.json" />
+    <None Include="tsconfig.json" />
+    <Content Update="wwwroot\**">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
     <None Update="NLog.config">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="wwwroot\js\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <TypeScriptCompile Include="scripts\ridgeLine.ts" />
+    <TypeScriptCompile Include="scripts\treeMap.ts" />
+    <TypeScriptCompile Include="scripts\types.ts" />
+    <TypeScriptCompile Include="scripts\utilities.ts" />
+  </ItemGroup>
+
+  <Target Name="JsBuild" BeforeTargets="PreBuildEvent">
+    <Exec Command="yarn build:dev" Condition="'$(Configuration)'=='Debug'" />
+    <Exec Command="yarn build:release" Condition="'$(Configuration)'=='Release'" />
+  </Target>
```

Comment:
It is highly undesirable to make yarn, hence node.js, a required tool to build Nethermind for a non-essential and optional component as the UI. The build failed for me just because I didn't have yarn.

A few suggestions:

- Do not use node.js/yarn at all
- Provide already minified versions (do we really need it, given its local usage only?)
- Having the UI as a separate NuGet package, like the health checks. I think this is the best approach and similar to the plugin approach suggested by @LukaszRozmej. I can help with that.

Reply:
node is only used in docker to build; its not included in the final container

Reply:
I agree that maintaining a NuGet package is somewhat annoying, but bringing the Node.js as a new build dependency for some optional components is also annoying. I believe not everyone in the team uses or even has Node.js & stuff, let alone third parties that build from source. Moreover, with a NuGet package, you'll have more liberties regarding the organization and approaches, as there will be much less nitpicking.
Of course, this is just my opinion, but I think I'm not alone in that. I'd like more opinions here.